### DISCUSSION TITLE
In what situations should should "subfuns" be non-empty?

### DISCUSSION THREAD

**Discussion by davisyoshida** [jax-ml/jax#16351] [Q&A] [answered]
For what JAX primitives can the first return value from `get_bind_params` be non-empty? I.e. `subfuns` below:

```python
subfuns, bind_params = primitive.get_bind_params(params)
```

For the primitives I've inspected it for it's always been empty. When it's non-empty, what are the contents?

**Comment by jakevdp** [comment, marked as answer]
This will basically only be non-empty for higher-order primitives, e.g. call primitives and custom jvp/vjp. If you search the JAX source code for classes that overload this method, you should be able to find all the primitives that return something non-trivial.

**Reply by davisyoshida** [reply]
Will do, thanks!
